test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +109,43 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+#
+# Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+#
+# Added as an effort to improve error handling and avoid full anyhow backtrace
+# https://github.com/neondatabase/neon/pull/7949
+#
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    env = neon_env_builder.init_start()
+    endpoint = env.endpoints.create_start("main")
+
+    tenant_id = env.initial_tenant
+    timeline_id = env.initial_timeline
```

Comment:
I would avoid this in new tests; I've sprinkled them around the test suite as previously there was a common method of always branching a timeline and testing that, in an effort to have a single pageserver serve multiple test cases. the python support for that was never realized with multiple processes, and it would be incompatible with global failpoints as well.

so, I'd just use these directly in a new test.